ebill_postfinance/models/ebill_postfinance_service.py

```diff
@@ -0,0 +1,153 @@
+# Copyright 2022 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+import logging
+import logging.config
+
+# from ..ebilling_postfinance.ebilling_postfinance import ebilling_postfinance
+from ebilling_postfinance import ebilling_postfinance
+
+from odoo import api, fields, models
+from odoo.exceptions import UserError
+
+_logger = logging.getLogger(__name__)
+
+
+class EbillPostfinanceService(models.Model):
+    _name = "ebill.postfinance.service"
+    _description = "Postfinance eBill service configuration"
+
+    name = fields.Char(required=True)
+    username = fields.Char()
+    password = fields.Char()
+    biller_id = fields.Char(string="Biller ID", size=17, required=True)
+    use_test_service = fields.Boolean(string="Testing", help="Target the test service")
+    partner_bank_id = fields.Many2one(
+        comodel_name="res.partner.bank", string="Bank account", ondelete="restrict"
+    )
+    invoice_message_ids = fields.One2many(
+        comodel_name="ebill.postfinance.invoice.message",
+        inverse_name="service_id",
+        string="Invoice Messages",
+        readonly=True,
+    )
+    ebill_payment_contract_ids = fields.One2many(
+        comodel_name="ebill.payment.contract",
+        inverse_name="postfinance_service_id",
+        string="Contracts",
+        readonly=True,
+    )
+    active = fields.Boolean(default=True)
+    file_type_to_use = fields.Selection(
+        string="Invoice Format",
+        default="XML",
+        required=True,
+        selection=[
+            ("XML", "XML Yellow Bill"),
+            ("EAI.XML", "Custom XML (SAPiDoc)"),
+            # ("eai.edi", "Custom EDIFACT"),
+            ("struct.pdf", "Factur X"),
+        ],
+    )
+    use_file_type_xml_paynet = fields.Boolean(
+        string="Use Paynet/SIX format",
+        help="Enable use of legacy SIX/Paynet invoice format.",
+    )
+    operation_timeout = fields.Integer(
+        string="HTTP Timeout",
+        default="600",
+        help="Timeout for each HTTP (GET, POST) request in seconds.",
+    )
+
+    def _get_service(self):
```

Comment:
Does it create a new session each time, can't we reuse a session by storing the current service?

Reply:
This is a very good point. And yes I believe there is some improvement to be done on this...